pyomo/contrib/community_detection/detection.py

```diff
@@ -0,0 +1,643 @@
+"""
+Main module for community detection integration with Pyomo models.
+
+This module separates model components (variables, constraints, and objectives) into different communities
+distinguished by the degree of connectivity between community members.
+
+Original implementation developed by Rahul Joglekar in the Grossmann research group.
+
+"""
+from logging import getLogger
+
+from pyomo.common.dependencies import attempt_import
+from pyomo.core import ConcreteModel, ComponentMap, Block, Var, Constraint, Objective, ConstraintList
+from pyomo.core.base.objective import _GeneralObjectiveData
+from pyomo.core.expr.current import identify_variables
+from pyomo.core.expr.visitor import replace_expressions
+from pyomo.contrib.community_detection.community_graph import generate_model_graph
+from pyomo.common.dependencies import networkx as nx
+from pyomo.common.dependencies import matplotlib, matplotlib_available
+from itertools import combinations
+
+import copy
+
+logger = getLogger('pyomo.contrib.community_detection')
+
+# Attempt import of louvain community detection package
+community_louvain, community_louvain_available = attempt_import(
+    'community', error_message="Could not import the 'community' library, available via 'python-louvain' on PyPI.")
+
+# Import matplotlib
+plt = matplotlib.pyplot
+cm = matplotlib.cm
```

Comment:
Remove this import with 
`color_map = plt.cm.get_cmap('viridis', len(numbered_community_map))`
which is done here 
https://github.com/Pyomo/pyomo/blob/2a09bf8b6b6f364a6d7a37a8f45b8079989356f2/pyomo/contrib/parmest/graphics.py#L162